src/hooks/useSearch.jsx

```diff
@@ -48,11 +48,23 @@ const keywordCategory = (product, word) => {
   }
 };
 // name에 대한 로직
-const keywordName = (product, word) => product.name.includes(word);
+const keywordName = (product, word) => {
+  const name = product.name.includes(word);
+  if (name) {
```

Comment:
이 부분 아래 keywordData 함수에서도 체크를 해서 keywordName 함수에서는 return만 해주면 될 것 같습니다!
```suggestion
  return name;
```
```js
if (nameList) {
    return nameList;
  } else {
    return categoryList;
  }
```